### PR TITLE
KAFKA-17609:[2/4]Convert system tests to kraft part 2

### PR DESCRIPTION
**Pull request by bbejeck**
part 2 of a multi-PR 

1. [Cooperative rebalance test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=bef928b5-ac66-371f-811a-273b636f0031)
2. [Named repartition test ]( https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=7b82d624-7df8-3183-9ca5-0f5d291adf5f)

### CODE REVIEW

**Comment by bbejeck** [comment]
Links to passing system test build

[Cooperative rebalance test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=bef928b5-ac66-371f-811a-273b636f0031)
[Named repartition test](https://semaphore.ci.confluent.io/jobs/8d9c8323-c32a-40f8-8d83-909743a86120/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976&suite_id=7b82d624-7df8-3183-9ca5-0f5d291adf5f)

File: tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def setUp(self):
             self.zk.start()
 
     @cluster(num_nodes=7)
-    @matrix(metadata_quorum=[quorum.isolated_kraft], use_new_coordinator=[True, False])
+    @matrix(metadata_quorum=[quorum.combined_kraft], use_new_coordinator=[True, False])
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py]: Why this change? Similar below.

Guess it raises a broader question: should we run with combined or isolate mode? IIRC, at some point in the past when we added KRaft support, we did mix it up on purpose to test both setups.

Frankly not sure, what we should do 🤔

**Review by mjsax — COMMENTED** [review]
Overall LGTM. Just not sure which KRaft mode we should use to run system tests...

File: tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py (hunk 1 above)
Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py]: My apologies!   I should have explained my change rationale up front.   I updated the `metadata_quorum` based on [this information in kafka.py](https://github.com/apache/kafka/blob/trunk/tests/kafkatest/services/kafka/kafka.py#L212-L235) In all of our system tests, we're using a combined cluster (not using dedicated nodes for the controller role) so I felt it best to use the value that represented what the test uses.  I could have been wrong in this approach/assumption so I'm willing to revert where necessary

File: tests/kafkatest/tests/streams/streams_cooperative_rebalance_upgrade_test.py [hunk 2]
```diff
@@ -68,9 +67,8 @@ def __init__(self, test_context):
                                            acks=1)
 
     @cluster(num_nodes=8)
-    @matrix(upgrade_from_version=streams_eager_rebalance_upgrade_versions)
-    def test_upgrade_to_cooperative_rebalance(self, upgrade_from_version):
-        self.zookeeper.start()
+    @matrix(upgrade_from_version=streams_eager_rebalance_upgrade_versions, metadata_quorum=[quorum.combined_kraft])
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_cooperative_rebalance_upgrade_test.py:70]: Seem we could remove `metadata_quorum` as test parameter?

Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_cooperative_rebalance_upgrade_test.py:70]: Right now the `metadata_quorum` in the `@matrix` annotation is required by test_context in `quorum.py` to determine if its using ZK or KRaft (maybe it'll pick it up from the parameter itself?) The `upgrade_from_version` points to an array so it doesn't have one value.

Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_cooperative_rebalance_upgrade_test.py:70]: Required to indicate to use KRaft for metadata quorum.  Failing to do so without a `@matrix` annotation results in this error as the test will default to ZK without it.
```File "/opt/kafka-dev/tests/kafkatest/services/kafka/kafka.py", line 541, in security_config
    zk_sasl=self.zk.zk_sasl if self.quorum_info.using_zk else False, zk_tls=self.zk_client_secure,
AttributeError: 'NoneType' object has no attribute 'zk_sasl'
```

**Review by mjsax — APPROVED** [review]
LGTM.

**Comment by bbejeck** [comment]
[Link to passing tests](https://semaphore.ci.confluent.io/jobs/fa50bfdc-920b-477e-93c2-90d9d57e18ca/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976)

zk_to_kraft_conversion_pr_2_results

**Comment by bbejeck** [comment]
Merged #17321 into trunk
